src/core-sdk-tasks/GetDependencyInfo.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    /// <summary>
+    /// Gets version and commit of a dependency by its name
+    /// from eng/Version.Details.xml
+    /// </summary>
+    public class GetDependencyInfo : Task
+    {
+        [Required]
+        public string DotnetInstallerCommit { get; set; }
+
+        [Required]
+        public string DependencyName { get; set; }
+
+        [Output]
+        public string DependencyVersion { get; set; }
+
+        [Output]
+        public string DependencyCommit { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                string versionsXmlUrl = $"https://raw.githubusercontent.com/dotnet/installer/{DotnetInstallerCommit}/eng/Version.Details.xml";
```

Comment:
Why are we making a web request here? Why aren't we just using the version from the local `Version.Details.xml` file?

Reply:
Good point! Fixed